components/ntask/org.wso2.carbon.ntask.core/src/main/java/org/wso2/carbon/ntask/core/impl/clustered/ClusteredTaskManager.java

```diff
@@ -47,59 +55,46 @@ public String getTaskType() {
         return this.getTaskRepository().getTasksType();
     }
 
-    public ClusterGroupCommunicator getClusterComm() throws TaskException {
-        return ClusterGroupCommunicator.getInstance(this.getTaskType());
-    }
-
+    /**
+     * {@inheritDoc}
+     */
     public void initStartupTasks() throws TaskException {
-        if (this.isLeader()) {
-            this.scheduleMissingTasks();
-        }
+        cleanupLeftoverTasks();
```

Comment:
Do we need this method? Can't we have the content of `cleanupLeftoverTasks()` here?

Reply:
A separate method was introduced to better reflect what was actually being done here (cleaning up). initStartupTasks is a method inherited from the TaskManager interface. Wouldn't it be clearer to have them as two separate methods?